File: typescript-automation/src/driver/ports/httpPort.ts
export interface HttpPort {
  getAs<T>(path: string): Promise<T>;
  getListAs<T>(path: string): Promise<ReadonlyArray<T>>;
  getListWithQueryAs<T>(path: string, paramName: string, paramValue: string): Promise<ReadonlyArray<T>>;
}

File: python-automation/src/automation/driver/ports/http_port.py
from abc import ABC, abstractmethod
from typing import Any


class HttpPort(ABC):

    @abstractmethod
    def get_as(self, path: str) -> dict[str, Any]: ...

    @abstractmethod
    def get_list_as(self, path: str) -> list[dict[str, Any]]: ...

    @abstractmethod
    def get_list_with_query_as(
        self, path: str, param_name: str, param_value: str
    ) -> list[dict[str, Any]]: ...

    @abstractmethod
    def close(self) -> None: ...
